### PR TITLE
Add Multizone deployment with global CP

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: ruromero <rromerom@redhat.com>

### CODE REVIEW

File: global-control-plane.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+```bash
+oc patch deploy -n kong kong-kong -p "{\"spec\": { \"template\" : { \"spec\" : {\"containers\":[{\"name\":\"proxy\",\"env\": [{ \"name\" : \"KONG_ADMIN_API_URI\", \"value\": \"$(oc get route -n kong kong-kong-admin -ojsonpath='{.spec.host}')\" }]}]}}}}"
+```
+
+- save the cluster/clustertelemetry endpoints for later
+
+```bash
+export CLUSTER_URL=$(oc get svc kong-kong-cluster -ojson | jq -r '.status.loadBalancer.ingress[].hostname')
+export CLUSTER_TELEMETRY_URL=$(oc get svc kong-kong-clustertelemetry -ojson | jq -r '.status.loadBalancer.ingress[].hostname')
```
[the remark this answers is not in the document]
Comment (mpaulgreen) [reply global-control-plane.md:77]: @ruromero This seems to be an issue with my openshift. All loadbalancer services are in pending state. You can ignore this comment for the time.

File: global-control-plane.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+- check the management UI is working at:
+
+```bash
+oc get route kong-kong-manager --template='{{.spec.host}}'
```
Comment (mpaulgreen) [inline global-control-plane.md:83]: oc -n kong get route kong-kong-manager --template='{{.spec.host}}'

File: global-control-plane.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+- check the admin endpoint is available
+
+```bash
+http `oc get route kong-kong-admin --template='{{.spec.host}}'` | jq .version
```
Comment (mpaulgreen) [inline global-control-plane.md:89]: http `oc -n kong get route kong-kong-admin --template='{{.spec.host}}'` | jq .version

File: global-control-plane.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+Enable TLS for the default mesh
+
+```bash
+kubectl apply -f mesh/mtls.yaml -n kong-mesh-system
+```
+
+Expose the svc
+
+```bash
+oc create route passthrough kong-mesh-control-plane --service=kong-mesh-control-plane --port=https-api-server
```
Comment (mpaulgreen) [inline global-control-plane.md:114]: oc -n kong-mesh-system create route passthrough kong-mesh-control-plane --service=kong-mesh-control-plane --port=https-api-server

File: global-control-plane.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+Login to the `kong-dp1` cluster
+
+```bash
+kumactl install control-plane --cni-enabled --mode=zone --zone=kumadp1 --ingress-enabled --kds-global-address grpcs://${GLOBAL_SYNC} | oc apply -f -
+```
+
+Check the zone is discovered in the global. Connect to the global control plane cluster to get the route
+
+```bash
+$ http `oc get route kong-mesh-control-plane -n kong-mesh-system -ojson | jq -r .spec.host`/status/zones
```
Comment (mpaulgreen) [inline global-control-plane.md:142]: https --verify=false `oc get route kong-mesh-control-plane -n kong-mesh-system -o json | jq -r .spec.host`/status/zones

File: global-control-plane.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+- check clustering
+
+**VM installation**: Refer to [Verify clustering is working](./gateway-multizone/kong-vm.md#verify-clustering-is-working)
+
+```bash
+http `oc get route -n kong kong-kong-admin --template='{{ .spec.host }}'`/clustering/status
+```
+
```
Comment (mpaulgreen) [inline global-control-plane.md:206]: You need to use a tools like kubectx to change the context.

**Review by mpaulgreen — CHANGES_REQUESTED** [review]

File: global-control-plane.md (hunk 5 above)
Comment (ruromero) [reply global-control-plane.md:142]: Yes I discovered the `https` command later haha, thanks

File: global-control-plane.md (hunk 6 above)
Comment (ruromero) [reply global-control-plane.md:206]: Thanks for the suggestion

**Comment by ruromero** [comment]
Added magnanimo to benigno communication using TrafficPermissions

**Review by mpaulgreen — APPROVED** [review]

**Pull request merged**